lib/ansible/plugins/action/__init__.py

```diff
@@ -298,7 +311,7 @@ def _is_become_unprivileged(self):
         '''
         # if we don't use become then we know we aren't switching to a
         # different unprivileged user
-        if not self._play_context.become:
+        if self._connection.become is None:
```

Comment:
For consistency why not just keep this as `if not self._connection.become`?  The next change below this one effectively uses that.  We should try to be consistent.